### PR TITLE
add support for brownie

### PR DESCRIPTION
**Pull request by jnsdls**

### CODE REVIEW

File: src/core/builder/builder-base.ts [hunk 1]
```diff
@@ -25,14 +26,23 @@ export abstract class BaseBuilder implements IBuilder {
...
 
-    var files = readdirSync(startPath);
-    for (var i = 0; i < files.length; i++) {
-      var filename = join(startPath, files[i]);
+    const files = readdirSync(startPath);
+    for (let i = 0; i < files.length; i++) {
+      const filename = join(startPath, files[i]);
       //skip the actual thirdweb contract itself
       if (basename(filename, ".json") === "ThirdwebContract") {
+        logger.debug(
```
Comment (joaquim-verges) [inline src/core/builder/builder-base.ts]: do we need this log? seems unnecessary?

File: src/core/builder/brownie.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { execSync } from "child_process";
+import { cosmiconfigSync } from "cosmiconfig";
+import { existsSync, readFileSync, rmdirSync } from "fs";
+import { basename, join } from "path";
+
+export class BrownieBuilder extends BaseBuilder {
+  public async compile(options: CompileOptions): Promise<{
+    contracts: ContractPayload[];
+  }> {
+    const explorer = cosmiconfigSync("brownie");
```
Comment (joaquim-verges) [inline src/core/builder/brownie.ts]: what does this load exactly here? seems heavy

File: src/core/builder/brownie.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (options.clean) {
+      logger.info("Cleaning build directory");
+      existsSync(buildPath) && rmdirSync(buildPath, { recursive: true });
+    }
+
+    logger.info("Compiling...");
+    execSync("brownie compile");
+
+    const contractsPath = join(buildPath, "contracts/");
```
Comment (joaquim-verges) [inline src/core/builder/brownie.ts:31]: is it safe to hardcode`contracts/` here?

Comment (jnsdls) [reply src/core/builder/brownie.ts:31]: yes, it's always relative to the build path

File: src/core/builder/builder-base.ts (hunk 1 above)
Comment (jnsdls) [reply src/core/builder/builder-base.ts]: yeah prob fair

File: src/core/builder/brownie.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { logger } from "../helpers/logger";
+import { CompileOptions } from "../interfaces/Builder";
+import { ContractPayload } from "../interfaces/ContractPayload";
+import { BaseBuilder } from "./builder-base";
+import { execSync } from "child_process";
+import { existsSync, readFileSync, rmdirSync } from "fs";
+import { basename, join } from "path";
+import { parse } from "yaml";
```
Comment (joaquim-verges) [inline src/core/builder/brownie.ts:8]: nice

File: src/core/builder/builder-base.ts [hunk 5]
```diff
@@ -25,14 +26,20 @@ export abstract class BaseBuilder implements IBuilder {
...
+      const filename = join(startPath, files[i]);
       //skip the actual thirdweb contract itself
       if (basename(filename, ".json") === "ThirdwebContract") {
         continue;
       }
-      var stat = statSync(filename);
+      const stat = statSync(filename);
+
+      // brownie has a "depdendencies" directory *inside* the build directory, if we detect that we should skip it
+      if (stat.isDirectory() && basename(filename) === "dependencies") {
```
Comment (joaquim-verges) [inline src/core/builder/builder-base.ts:39]: hopefully this base class handling doesnt get crazier than this - otherwise might have to pull it up

**Review by joaquim-verges — APPROVED** [review]
lgtm

File: src/core/builder/builder-base.ts (hunk 5 above)
Comment (jnsdls) [reply src/core/builder/builder-base.ts:39]: yeh

**Pull request merged**